packages-tooling/__tests__/src/plugin-conventions/type-checking/without-convention.basic.spec.ts

```diff
@@ -2115,5 +2115,473 @@ ${prop('prop', 'string', isTs)}
+                   * @param {string} x
+                   * @param {number} y
+                   * @returns {string}
+                   */
+                  /**
+                   * @param {string | number} x
+                   * @param {number} y
+                   * @returns {string | number}
+                   */
+                  `}
+                ${isTs ? `public ` : ''}prop(x${isTs ? `: string | number` : ''}, y${isTs ? `: number` : ''})${isTs ? `: string | number` : ''} { return typeof x === 'number' ? x + y : \`\${x} \${y}\`; };
+                }
+                `,
+            readFile: createMarkupReader(markupFile, markup),
+          }, nonConventionalOptions);
+
+        assertSuccess(entry, result.code);
+      });
+
+      it(`overloaded method - fail`, function () {
+        const entry = `entry.${extn}`;
+        const markupFile = 'entry.html';
+        const markup = '${prop(\'foo\', 42).toExponential(2)} ${prop(3.14, 42).toUpperCase()}';
+        const result = preprocessResource(
+          {
+            path: entry,
+            contents: `
+                      import { customElement } from '@aurelia/runtime-html';
+                      import template from './${markupFile}';
+
+                      @customElement({ name: 'foo', template })
+                      export class Foo {
+                      ${prop('bar', 'number | boolean', isTs)}
+                      ${isTs ? `
+                        public prop(x: number, y: number): number;
+                        public prop(x: string, y: number): string;
+                        ` : `
+                        /**
+                         * @overload
+                         * @param {number} x
+                         * @param {number} y
+                         * @returns {number}
+                         */
+                        /**
+                         * @overload
+                         * @param {string} x
+                         * @param {number} y
+                         * @returns {string}
+                         */
+                        /**
+                         * @param {string | number} x
+                         * @param {number} y
+                         * @returns {string | number}
+                         */
+                        `}
+                      ${isTs ? `public ` : ''}prop(x${isTs ? `: string | number` : ''}, y${isTs ? `: number` : ''})${isTs ? `: string | number` : ''} { return typeof x === 'number' ? x + y : \`\${x} \${y}\`; };
+                      }
+                      `,
+            readFile: createMarkupReader(markupFile, markup),
+          }, nonConventionalOptions);
+
+        assertFailure(entry, result.code, [/Property 'toExponential' does not exist on type 'string'/, /Property 'toUpperCase' does not exist on type 'number'/]);
+      });
+    });
+
+    describe('access modifier', function () {
+      for (const am of ['public', 'protected', 'private'] as const) {
```

Comment:
nit
```suggestion
      for (const modifier of ['public', 'protected', 'private'] as const) {
```
😃 there's a lot of code in these files, not even having to jump once is a lot better.

Reply:
Changes done.